lib/std/math/powi.zig

```diff
@@ -10,108 +10,92 @@ const testing = std.testing;
 
 /// Returns the power of x raised by the integer y (x^y).
 ///
+/// Errors:
+///  - Overflow: Integer overflow or Infinity
+///  - Underflow: Absolute value of result smaller than 1
 /// Special Cases:
-///  - powi(x, +-0)   = 1 for any x
-///  - powi(0, y)     = 0 for any y
-///  - powi(1, y)     = 1 for any y
-///  - powi(-1, y)    = -1 for y an odd integer
-///  - powi(-1, y)    = 1 for y an even integer
-///  - powi(x, y)     = Overflow for y >= @sizeOf(x) - 1 or y > 0
-///  - powi(x, y)     = Underflow for y > @sizeOf(x) - 1 or y < 0
+///  - powi(x, 0)   = 1 unless T is i1, i0, u0
+///  - powi(0, x)   = 0
+///  - powi(0, x)   = Overflow when x < 0
+///  - powi(1, y)   = 1 unless T is i1, i0, u0
+///  - powi(-1, y)  = -1 for y an odd integer
+///  - powi(-1, y)  = 1 unless T is i1, i0, u0
+///  - powi(x, y)   = Overflow for y >= @bitSizeOf(x)
+///  - powi(x, y)   = Underflow for y < 0
 pub fn powi(comptime T: type, x: T, y: T) (error{
     Overflow,
     Underflow,
 }!T) {
-    const info = @typeInfo(T);
+    const bit_size = @bitSizeOf(T);
 
     comptime assert(@typeInfo(T) == .Int);
 
-    //  powi(x, +-0)   = 1 for any x
-    if (y == 0 or y == -0) {
-        return 1;
+    // integer value 1 cannot be coerced to type 'i1'
```

Comment:
I dont see the coercion here. please be more specific. As I understand it, you try to special case any operation on u0,i0,u1,i1.

However, `const t1: u1 = 0; const t2: u1 = 0;` with `0**0` and y being anything is a valid operation. And same for `i1`.

Reply:
i documented why it won't compile